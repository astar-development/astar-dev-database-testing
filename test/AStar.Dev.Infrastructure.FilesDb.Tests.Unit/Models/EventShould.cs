using AStar.Dev.Utilities;
using Shouldly;

namespace AStar.Dev.Infrastructure.FilesDb.Models;

public class EventShould
{
    [Fact]
    public void ShouldContainTheExpectedProperties()
        => new Event
           {
               FileName         = "FileName",
               DirectoryName    = "DirectoryName",
               FileSize         = 124,
               FileCreated      = new (2025, 07, 28, 11, 22, 33, TimeSpan.Zero) ,
               FileLastModified = new (2025, 07, 28, 1,  2,  3,  TimeSpan.Zero),
               Height           = 456,
               Id               = 1324,
               UpdatedBy        = "TestUser",
               Width            = 987,
               EventOccurredAt  = new (2025, 07, 28, 1, 24, 34, TimeSpan.Zero),
               Handle           = "mock-handle",
               Type             = EventType.Add
           }
           .ToJson()
           .ShouldMatchApproved();
}
