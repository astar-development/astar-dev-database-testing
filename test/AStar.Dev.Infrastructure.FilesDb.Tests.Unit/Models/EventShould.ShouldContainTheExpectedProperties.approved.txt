{
  "id": 1324,
  "type": {
    "value": 1,
    "name": "Add"
  },
  "eventOccurredAt": "2025-07-28T01:24:34+00:00",
  "fileName": "FileName",
  "directoryName": "DirectoryName",
  "handle": "mock-handle",
  "width": 987,
  "height": 456,
  "fileSize": 124,
  "fileCreated": "2025-07-28T11:22:33+00:00",
  "fileLastModified": "2025-07-28T01:02:03+00:00",
  "updatedBy": "TestUser"
}